lib/run-context.js

```diff
@@ -119,7 +119,9 @@ RunContext.prototype._run = function () {
  */
 RunContext.prototype.toPromise = function () {
   return new Promise(function (resolve, reject) {
-    this.on('end', resolve);
+    this.on('end', function () {
+      resolve(this.targetDirectory);
```

Comment:
But if you think that it makes more sense to bind it to `this`, I would understand